### DISCUSSION TITLE
Accept uppercase params

### DISCUSSION THREAD

**Discussion by fabry24** [developmentseed/titiler#489] [Q&A] [answered]
Hi,
It is possible to change some part of code, for accepting uppercase params, (URL, BIDX, RESCALE....)
I try to add wmts layer from titiler to geoserver, but geoserver sends uppercase parameters.

I thought that it is in src/titiler/core/titiler/core/factory.py in def wmts(), but i did not reach function wmts() in function wmts()

Thanks.

**Comment by vincentsarago** [comment, marked as answer]
This is already possible if you use the `LowerCaseQueryStringMiddleware` middleware introduced in https://github.com/developmentseed/titiler/pull/321

in the default titiler application you can set `TITILER_API_LOWER_CASE_QUERY_PARAMTERS=TRUE` environment to activate it
https://github.com/developmentseed/titiler/blob/73a8eb00bcf14151667579dd859b411ddd2636f4/src/titiler/application/titiler/application/main.py#L86-L87

**Reply by vincentsarago** [reply]
I've updated the docs https://github.com/developmentseed/titiler/blob/master/docs/src/intro.md#settings 🙏

**Comment by fabry24** [comment]
Thanks, I changed it in src\titiler\application\titiler\application\settings.py
